engine/crates/engine-v2/src/sources/graphql/request/execute.rs

```diff
@@ -68,11 +69,12 @@ pub(crate) async fn execute_subgraph_request<'ctx, 'a, R: Runtime>(
             timeout: endpoint.config.timeout,
         }
     };
-    ctx.record_request(&request);
 
     let fetcher = ctx.engine.runtime.fetcher();
+    let http_span = ctx.create_subgraph_request_span(&request);
```

Comment:
I would have created the HTTP span immediately with the context and only added `instrument` here. That way we would not need to have both
`http_span.record_http_status_code(response.status());` and `ctx.record_http_response(&response);`. Having a single "context" to record what happens and update the relevant spans & metrics feels simpler to me imho

Reply:
I know you don't have the http request when creating the span and need to put everything as `Empty`, but that feels still cleaner than having multiple things to call here to me IMHO. Don't have a strong opinion though.